compiler/cli/cli-common/src/org/jetbrains/kotlin/cli/common/messages/CompilerMessageExtendedLocation.kt

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright 2010-2020 JetBrains s.r.o. and Kotlin Programming Language contributors.
+ * Use of this source code is governed by the Apache 2.0 license that can be found in the license/LICENSE.txt file.
+ */
+
+package org.jetbrains.kotlin.cli.common.messages
+
+import java.io.Serializable
+
+interface CompilerMessageExtendedLocation {
+    val path: String
+    val line: Int
+    val column: Int
+    val lineEnd: Int
+    val columnEnd: Int
+    val lineContent: String?
```

Comment:
Is it possible to store `CompilerMessageLocation` + `lineEnd` + `columnEnd` here?

Reply:
The current approach seems to be more simple for usage and also more perfomant (because there is no one additional reference). Do you really want to change it?

Reply:
I'm just concerned that this is a copy-paste of the internals of `CompilerMessageLocation`, and semantically this class seems to be exactly `CompilerMessageLocation` + line/column end.

As an example, if somebody changes something in `CompilerMessageLocation` in the future, there will be no hint that the corresponding change is also needed here in `CompilerMessageExtendedLocation`. Which is a thing that happens with copy-pasted code all the time.

To keep usages simple, we might still expose these properties like this:
```
val line: Int get() = location.line
val column: Int get() = location.column
```

Reply:
Ok, agree.

Reply:
Fixed

Reply:
Thanks! Only now I noticed that `path`, `line`, `column`, `lineContent` do not actually need to be interface members. Can we extract them to extensions? I.e.
```
val CompilerMessageExtendedLocation.path: String
    get() = location.path
```

Reply:
Really so. Extracted :) Thank you